craftcms/cms
# Fix broken JS on Cloud uploader when changing dirs

Following https://github.com/craftcms/cms/pull/14439, a JS error was thrown when navigating directories.
This is because AssetIndex was calling `.fileupload()`, which can't be assumed, as only hte core Uploader uses the jquery uploader plugin.

## Changes
- When selecting source, make `onSelectSource` fire before `onSourcePathChange` to be less awkward.
- With `onSelectSource` fired first, we can be certain out uploader is created by the time we get to `onSourcePathChange`
- Don't call `fileupload` methods if the plugin wasn't instantiated (eg if uploading is disabled)
- Eliminate duplicate inputs by setting`replaceFileInput = false`, which is only needed for very old iframe uploading. 

## History
- this was reported: https://github.com/craftcms/cms/issues/14403
- first fix (@brandon): https://github.com/craftcms/cms/commit/af652dba5d7e58e001350c297963bdf0ff435ab5
- but then it caused this: https://github.com/craftcms/cms/issues/14438
- second fix (@i-just ): https://github.com/craftcms/cms/pull/14439

## Test cases
- permissions
  - create 2 volumes
  - user can do everything for vol1
  - user can only view vol2
  - vol1 has one subfolder
  - go to vol1 > subfolder & upload a file - all good
  - go to vol2 - upload button is disabled and you can’t upload - all good
  - go to vol1, you should be deep linked into the subfolder (cause that’s where you were last); try to upload a file
- Assets page
  - upload button
  - drag to upload
- Assets field
  - select via modal
  - upload directly
  - upload directly via drag
